examples/audio/sound_effects.rs

```diff
@@ -0,0 +1,207 @@
+        // This must be below the `DefaultPlugins` plugin, as it depends on the `AssetPlugin`.
+        .init_resource::<SoundEffects>()
+        .add_systems(Startup, spawn_button)
+        .add_systems(
+            Update,
+            (
+                play_sound_effect_when_button_pressed,
+                change_button_color_based_on_interaction,
+            ),
+        )
+        .run();
+}
+
+#[derive(Resource)]
+struct SoundEffects {
+    map: HashMap<String, Vec<Handle<AudioSource>>>,
+}
+
+impl SoundEffects {
+    /// Plays a random sound effect matching the given name.
+    ///
+    /// When defining the settings for this method, you almost always want to use [`PlaybackMode::Despawn`](bevy::audio::PlaybackMode).
+    /// Every time a sound effect is played, a new entity is generated. Once the sound effect is complete,
+    /// the entity should be cleaned up, rather than looping or sitting around uselessly.
+    ///
+    /// This method accepts any type which implements `AsRef<str>`.
+    /// This allows you to pass in `&str`, `String`, or a custom type that can be converted to a string.
+    ///
+    /// These custom types can be useful for defining enums that represent specific sound effects.
+    /// Generally speaking, enum values should be used to represent one-off or special-cased sound effects,
+    /// while string keys are a better fit for sound effects corresponding to objects loaded from a data file.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// enum GameSfx {
+    ///     SplashScreenJingle,
+    ///     Victory,
+    ///     Defeat,
+    /// }
+    ///
+    /// impl AsRef<str> for GameSfx {
+    ///     fn as_ref(&self) -> &str {
+    ///         match self {
+    ///             GameSfx::SplashScreenJingle => "splash_screen_jingle",
+    ///             GameSfx::Victory => "victory",
+    ///             GameSfx::Defeat => "defeat",
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    fn play(&mut self, name: impl AsRef<str>, world: &mut World, settings: PlaybackSettings) {
+        let name = name.as_ref();
+        if let Some(sfx_list) = self.map.get_mut(name) {
+            // If you need precise control over the randomization order of your sound effects,
+            // store the RNG as a resource and modify these functions to take it as an argument.
+            let rng = &mut rand::thread_rng();
+
+            let index = rng.sample(Uniform::from(0..sfx_list.len()));
+            // We don't need a (slightly) more expensive strong handle here (which are used to keep assets loaded in memory)
+            // because a copy is always stored in the SoundEffects resource.
+            let source = sfx_list[index].clone_weak();
+
+            world.spawn(AudioBundle {
+                source,
+                // We want the sound effect to play once and then despawn.
+                settings,
+            });
+        } else {
+            warn!("Sound effect not found: {}", name);
```

Comment:
@janhohenheim tells me to recommend this :)

```suggestion
            warn!("Sound effect not found: {name}");
```